### PR TITLE
Feature/pass fail

### PR DESCRIPTION
**Pull request by gweitkemper**
Couldn't sleep so I took a look at this.
There's some repeating code that I'd extract as a function to make life easier.
That if statement might be better as a switch statement.

I've made both changes on my local copy and can commit it if you'd like.
Which way would be a more effective learning strategy for you?  Leave you with the hints above or show the examples in my code?

### CODE REVIEW

**Comment by builtauthentic** [comment]
Show 'em, if you got 'em. I'll look over the compare and see where you're coming from.

**Comment by gweitkemper** [comment]
My local blew up so I had to copy/paste the whole file so it looks more dramatic than it is.  I hope it's not too hard to follow.  If so I can kill my changes and redo them

**Comment by builtauthentic** [comment]
Let's talk through the changes tonight, if you have time. I can see the benefit of some, but I probably need to evolve how I'm thinking about the structure a bit more and going over pros/cons of either solution might help.

**Comment by gweitkemper** [comment]
Sure - I'm free from 4:30-8 and from 9-???.

High level - Pro of extracting the reused code to a func is that it makes it reusable.  Generally anything you type 3 times, you should probably find a way to make repeatable.
For the switch case, it just makes it less forks in logic, granted switches aren't great I just don't know if there's a better way in Go

Cons are that from a learning perspective doing everything by hand is easier to learn especially at first.  

Looking forward to catching up.
